### DISCUSSION TITLE
How do I turn text horizontal/vertical ?

### DISCUSSION THREAD

**Discussion by MarcusE1W** [JakeStanger/ironbar#294] [Q&A] [answered]
I am using the example config with one modification:
```
 "position": "left",
```
After this the text for sys_info gets turned 90 degrees. This creates very long output inside the bar.
How can I change the orientation so that it flows from left to right instead from top to bottom?

**Comment by JakeStanger** [comment]
Currently it is not possible to override the text direction, as this is just determined based on the bar position. 

I think adding a new config option for this should be fairly trivial though, so please feel free to open a feature request issue .

**Comment by MarcusE1W** [comment]
I think to write it myself is a bit over my rust knowledge and it also involves a design decision how to handle this in general.
The 90 degree turn mostly affects `label`, `sys_info`and `clock`?
As a starting position I would say that nothing needs to be turned. 

Interestingly enough `label` actually does not get turned. It is missing the a call to `label.set_angle(info.bar_position.get_angle());`. This might actually  be by mistake ?

The `custom`module has a setting for `orientation`, maybe that could be used for the other modules as well?

An easy fix would be to just take out the lines that turn `clock` and `sys_info`. I am just not quite sure if `sys_info` would use several lines per information or just show one very long single line.

Alternatively all relevant modules could get the parameter `orientation` ?

**Reply by JakeStanger** [reply, marked as answer]
Don't worry if you can't write it yourself, I just need it logged in the issues so it can be tracked. You're correct the rotate call in `label`  has been missed.

I'm reluctant to take out the rotation code in the modules already unless there's a general consensus that would be better, as that would be a breaking change to anybody currently happily using a vertical bar. The likely solution is I'll add a new property to specify the text angle.

An overlapping issue #296 has been opened, which we can use to track this.
